src/app/containers/NewStoryPromo/PromoOrganisms/RelatedContentPromo/index.jsx

```diff
@@ -0,0 +1,91 @@
+import React, { useContext } from 'react';
+import { shape, arrayOf, string } from 'prop-types';
+import SectionLabel from '#legacy/psammead-section-label/src';
+import path from 'ramda/src/path';
+import pathOr from 'ramda/src/pathOr';
+import pathEq from 'ramda/src/pathEq';
+import tail from 'ramda/src/tail';
+import slice from 'ramda/src/slice';
+import identity from 'ramda/src/identity';
+import { ServiceContext } from '#app/contexts/ServiceContext';
+import {
+  StoryPromoUlGrid,
+  StyledStoryPromoLi,
+  StyledWrapper,
+  SingleItemWrapper,
+  SingleItemGrid,
+} from './index.styles';
+import RelatedContentItem from './RelatedContentItem';
+
+const RelatedContentPromo = ({ content }) => {
+  const { translations } = useContext(ServiceContext);
+
+  if (!pathEq(['type'], 'relatedContent', content)) return null;
+
+  if (!content) return null;
+  const items = path(['model', 'blocks'], content);
+
+  const LABEL_ID = 'related-content-heading';
+
+  const customTitle =
+    pathEq([0, 'type'], 'title', items) &&
+    path(
+      [0, 'model', 'blocks', 0, 'model', 'blocks', 0, 'model', 'text'],
+      items,
+    );
+
+  const title =
+    customTitle || pathOr('Related Content', ['relatedContent'], translations);
+
+  if (customTitle) tail(items);
+
+  const storyPromoItems = customTitle ? tail(items) : identity(items);
+
+  const reducedStoryPromoItems = slice(0, 6, storyPromoItems);
+
+  const hasSingleContent = reducedStoryPromoItems.length === 1;
+
+  return (
+    <StyledWrapper aria-labelledby={LABEL_ID} role="region" data-e2e={LABEL_ID}>
```

Comment:
Still struggling with the `RelatedContentPromo` component rendering a _list_ of `RelatedContentPromo`s in some circumstances. If that is the approach, should this component be called `RelatedContentSection` to be clear it is rendering a full section including either a single item or a list of `RelatedContentItems`. Just makes more sense to my brain 🤔

Reply:
Yeah, it makes sense to rename it RelatedContentSection. I will do that. I believe this applies to TopStoriesPromo. I believe that can become TopStoriesSection as well.